doc/api/http.md

```diff
@@ -157,6 +157,33 @@ socket/stream from this function, or by passing the socket/stream to `callback`.
 
 `callback` has a signature of `(err, stream)`.
 
+### agent.keepSocketAlive(socket)
+<!-- YAML
+added: REPLACEME
+-->
+
+Called when socket is detached from a request and could be persisted by the
+Agent. Default behavior is to:
+
+* `socket.unref()`
+* `socket.setKeepAlive()` with the `agent.keepAliveMsecs` as an argument
+
+This method can be overridden by particular `Agent` class child. If this method
+returns `false`y value - the socket will be destroyed instead of persisting it
+for use with the next request.
+
+### agent.reuseSocket(socket, req)
+<!-- YAML
+added: REPLACEME
+-->
+
+Called when socket is attached to request after being persisted because of the
+keep-alive options. Default behavior is to:
+
+* `socket.ref()`
```

Comment:
I'm not sure this makes sense to have as a bulleted list. I think just having it inline is good enough.